driver_test.go

```diff
@@ -1038,6 +1015,51 @@ func TestPodmanDriver_NetworkMode(t *testing.T) {
 	}
 }
 
+// test kill / signal support
+func TestPodmanDriver_SignalTask(t *testing.T) {
+	if !tu.IsCI() {
+		t.Parallel()
+	}
+
+	taskCfg := newTaskConfig("", busyboxLongRunningCmd)
+	task := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "signal_task",
+		AllocID:   uuid.Generate(),
+		Resources: createBasicResources(),
+	}
+	require.NoError(t, task.EncodeConcreteDriverConfig(&taskCfg))
+
+	d := podmanDriverHarness(t, nil)
+	cleanup := d.MkAllocDir(task, true)
+	defer cleanup()
+
+	_, _, err := d.StartTask(task)
+	require.NoError(t, err)
+
+	defer d.DestroyTask(task.ID, true)
+
+	go func(t *testing.T) {
```

Comment:
I'm not sure if we can safely use require methods and `t` in a goroutine

```go
errCh := make(chan error, 2)
go func() {
        err := d.SignalTask(task.ID, "FOO")
        errCh <- err
        err := d.SignalTask(task.ID, "SIGINT")
        errCh <- err
}()
```

Then you can select twice on errCh or timeout below, also time.Sleeps should probably be removed in favor of retrying, we can copy over nomads test helper for this. https://github.com/hashicorp/nomad/blob/5e34efadab66de20d1a64691916f10df2e8255f8/api/internal/testutil/wait.go#L11

Reply:
Not sure why the go func() was introduced. It's not necessary for this test. I removed it and verified that the test indeed works.